packages/tools/replay-tool/src/replayMessages.ts

```diff
@@ -230,10 +238,8 @@ class Document {
         }
     }
 
-    public async snapshot() {
-        return (this.container as Container).snapshot(
-            `ReplayTool Snapshot: op ${this.currentOp}, ${this.getFileName()}`,
-            !this.args.incremental /* generateFullTreeNoOtimizations */);
+    public async summarize() {
+        await uploadContainerSummary(this.container);
```

Comment:
Sharing some thoughts I have had with this pattern and testing summaries in general. It doesn't have to be in this PR as I think that would take lot more work:
It would be a good idea to get the summary tree here directly and use it rather than going via the driver layer. The challenge would be getting the protocol tree.
Also, it would be great if we could load a container from a summary without going through so many hoops and layers.

Reply:
Yes - we chatted about this earlier. After debating a bit, I thought this would be a better route since eventually we want to call a function like `submitSummary` that may not necessarily be returning the summary back to the API caller. Also, here we are validating present (and future) layers below container, which somewhat made sense in the context of "lets test summaries/snapshots" vs "lets test container functionality". 

I'll create follow-up ticket on this topic. There are few more related items.

Reply:
Referred to this convo here: https://github.com/microsoft/FluidFramework/issues/8973. Let's use that ticket to capture testability improvements as well.